docs/terraform-asm-mcp/gke.tf

```diff
@@ -41,6 +41,7 @@ module "gke" {
   enable_private_nodes    = false
   master_ipv4_cidr_block  = "172.16.0.0/28"
   cluster_resource_labels = { "mesh_id" : "proj-${data.google_project.project.number}" }
+  deletion_protection     = false # Warning: Do not set deletion_protection to false for production clusters
```

Comment:
This module (which uses google_container_cluster under the hood) set `deletion_protection` to false. See [related module code here](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-private-cluster-update-variant/variables.tf).